packages/beacon-node/src/chain/produceBlock/produceBlockBody.ts

```diff
@@ -338,4 +352,10 @@ async function getExecutionPayloadParentHash(
   }
 }
 
+function getExpectedWithdrawals(state: CachedBeaconStateCapella): capella.Withdrawal[] {
+  const numWithdrawals = Math.min(MAX_WITHDRAWALS_PER_PAYLOAD, state.withdrawalQueue.length);
+  const withdrawals = state.withdrawalQueue.toValue();
```

Comment:
This is very inefficient, you are reading converting an unbounded list of max length 2*40, to a value; then slicing a tiny portion. Instead create a new array and iterate what you need